nodejs/eks/nodegroup.ts

```diff
@@ -310,17 +315,50 @@ ${customUserData}
 `;
         });
 
-    let amiId: pulumi.Input<string> = args.amiId!;
+    let amiId: any = (args.amiId) ? args.amiId : undefined;
```

Comment:
I don’t follow why this needed to change - and generally this could be simpler (in fact even the original code is more complicated than needed - the `!` should have been unnecessary). 

But most importantly, we should not use `: any` unless absolutely required - it throws away valuable type checking.

Reply:
IIRC using the `!` resulted in subsequent errors about the var being undefined or a promise and this not being able to set to the vars type. I'll look into it, but any other suggestions?

Reply:
This could be fixed - but happy to do that later.